subsys/net/l2/wifi/wifi_shell.c

```diff
@@ -1255,6 +1263,335 @@ static int cmd_wifi_ps_wakeup_mode(const struct shell *sh, size_t argc, char *ar
 	return 0;
 }
 
+void parse_mode_args_to_params(const struct shell *sh, int argc,
+			       char *argv[], struct wifi_mode_info *mode,
+			       bool *do_mode_oper)
+{
+	int opt;
+	int option_index = 0;
+
+	static struct option long_options[] = {{"if_index", optional_argument, 0, 'i'},
+					       {"sta", no_argument, 0, 's'},
+					       {"monitor", no_argument, 0, 'm'},
+					       {"TX-injection", no_argument, 0, 't'},
+					       {"promiscuous", no_argument, 0, 'p'},
+					       {"ap", no_argument, 0, 'a'},
+					       {"softap", no_argument, 0, 'k'},
+					       {"get", no_argument, 0, 'g'},
+					       {"help", no_argument, 0, 'h'},
+					       {0, 0, 0, 0}};
+
+	while ((opt = getopt_long(argc, argv, "i:smtpakgh", long_options, &option_index)) != -1) {
+		switch (opt) {
+		case 's':
+			mode->mode |= WIFI_STA_MODE;
+			break;
+		case 'm':
+			mode->mode |= WIFI_MONITOR_MODE;
+			break;
+		case 't':
+			mode->mode |= WIFI_TX_INJECTION_MODE;
+			break;
+		case 'p':
+			mode->mode |= WIFI_PROMISCUOUS_MODE;
+			break;
+		case 'a':
+			mode->mode |= WIFI_AP_MODE;
+			break;
+		case 'k':
+			mode->mode |= WIFI_SOFTAP_MODE;
+			break;
+		case 'g':
+			mode->oper = true;
+			break;
+		case 'i':
+			mode->if_index = (uint8_t)atoi(optarg);
+			break;
+		case 'h':
+			shell_help(sh);
+			*do_mode_oper = false;
+			break;
+		case '?':
+		default:
+			break;
+		}
+	}
+}
+
+static int cmd_wifi_mode(const struct shell *sh, size_t argc, char *argv[])
+{
+	struct net_if *iface = NULL;
```

Comment:
No need to set the iface to null as we always set a value to it later in this function.